src/drivers/amd/renoir/interrupt.c

```diff
@@ -0,0 +1,386 @@
+	platform_interrupt_set(IRQ_NUM_EXT_LEVEL5);
+	platform_interrupt_set(IRQ_NUM_EXT_LEVEL4);
+}
+
+void acp_intr_disable(void)
+{
+	acp_dsp0_intr_stat_t interruptstatus;
+	acp_external_intr_enb_t extinterruptenb;
+	acp_dsp0_intr_cntl_t interruptcntl;
+
+	interruptstatus = (acp_dsp0_intr_stat_t) io_reg_read((PU_REGISTER_BASE +
+							ACP_DSP0_INTR_STAT));
+	/*Check and Clear all the Interrupt status bits*/
+	interruptstatus.bits.dmaiocstat			= 0xFF;
+	interruptstatus.bits.audio_buffer_int_stat	= 0x3F;
+	interruptstatus.bits.wov_dma_stat		= INTERRUPT_ENABLE;
+	io_reg_write((PU_REGISTER_BASE + ACP_DSP0_INTR_STAT), interruptstatus.u32all);
+	/*Disable the ACP to Host interrupts*/
+	extinterruptenb.bits.acpextintrenb = INTERRUPT_CLEAR;
+	io_reg_write((PU_REGISTER_BASE + ACP_EXTERNAL_INTR_ENB),  extinterruptenb.u32all);
+	/*Disable all the required ACP interrupts in acp_dsp0_intr_cntl_t register*/
+	interruptcntl = (acp_dsp0_intr_cntl_t) io_reg_read((PU_REGISTER_BASE + ACP_DSP0_INTR_CNTL));
+	interruptcntl.bits.dmaiocmask			= INTERRUPT_DISABLE;
+	interruptcntl.bits.audio_buffer_int_mask	= INTERRUPT_DISABLE;
+	interruptcntl.bits.wov_dma_intr_mask		= INTERRUPT_DISABLE;
+	io_reg_write((PU_REGISTER_BASE + ACP_DSP0_INTR_CNTL),  interruptcntl.u32all);
+	platform_interrupt_clear(IRQ_NUM_EXT_LEVEL5, 0);
+	platform_interrupt_clear(IRQ_NUM_EXT_LEVEL3, 0);
+	platform_interrupt_clear(IRQ_NUM_EXT_LEVEL4, 0);
+	platform_interrupt_clear(IRQ_NUM_TIMER1, 0);
+	platform_interrupt_clear(IRQ_NUM_TIMER0, 0);
+}
+
+void acp_dsp_sw_intr_disable(void)
+{
+	acp_dsp_sw_intr_cntl_t  swintrctrlreg;
+	/* Read the register */
+	swintrctrlreg = (acp_dsp_sw_intr_cntl_t) io_reg_read((PU_REGISTER_BASE +
+							ACP_DSP_SW_INTR_CNTL));
+	swintrctrlreg.bits.dsp0tohostintrmask = INTERRUPT_DISABLE;
+	/* Write the Software Interrupt controller register */
+	io_reg_write((PU_REGISTER_BASE + ACP_DSP_SW_INTR_CNTL), swintrctrlreg.u32all);
+	platform_interrupt_clear(IRQ_NUM_EXT_LEVEL3, 0);
+}
+
+/*This function triggers a host interrupt from ACP DSP*/
+void acp_dsp_to_host_Intr_trig(void)
+{
+	acp_sw_intr_trig_t  swintrtrig;
+	/* Read the Software Interrupt controller register and update */
+	swintrtrig = (acp_sw_intr_trig_t) io_reg_read((PU_REGISTER_BASE + ACP_SW_INTR_TRIG));
+	/*Configures the trigger bit in ACP_DSP_SW_INTR_TRIG register*/
+	swintrtrig.bits.trigdsp0tohostintr = INTERRUPT_ENABLE;
+	/* Write the Software Interrupt controller register */
+	io_reg_write((PU_REGISTER_BASE + ACP_SW_INTR_TRIG), swintrtrig.u32all);
+}
+
+/* Clear the Acknowledge ( status) for the host to DSP inteerupt */
+void acp_ack_intr_from_host(void)
+{
+	/* acknowledge the host interrupt*/
+	acp_dsp_sw_intr_stat_t swintrstat;
+
+	swintrstat = (acp_dsp_sw_intr_stat_t)io_reg_read((PU_REGISTER_BASE +
+							ACP_DSP_SW_INTR_STAT));
+	swintrstat.bits.hosttodsp0intr1stat = INTERRUPT_ENABLE;
+	io_reg_write((PU_REGISTER_BASE + ACP_DSP_SW_INTR_STAT), swintrstat.u32all);
```

Comment:
in fact there are several locations like this: a comma has the lowest priority in C, so, no need for parentheses around complete function parameters